File: 2022/01/main.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	content, _ := os.ReadFile("input.txt")
	s_content := string(content)
	temp := strings.Split(s_content, "\n")

	var count int
	count = 0

	var totals []int

	for _, value := range temp {
		s_value := string(value)

		if s_value == "" {
			totals = append(totals, count)
			count = 0
		}

		int, _ := strconv.Atoi(s_value)
		count += int
	}
	sort.Ints(totals)

	lastElement := totals[len(totals)-1]
	topThree := totals[len(totals)-1] + totals[len(totals)-2] + totals[len(totals)-3]

	fmt.Println("Part 1: ", lastElement)
	fmt.Println("Part 2: ", topThree)
}

func printSlice(s []int) {
	fmt.Printf("len=%d cap=%d %v\n", len(s), cap(s), s)
}

File: 2022/01/main.py
with open("input.txt") as file:
    input = file.readlines()

count = 0
elf = {}

for i in input:
    i = i.strip()

    if i == '\n' or i == '':
        count += 1
        continue

    current = elf.get(count, 0)
    current += int(i)
    elf[count] = current

print("Part 1")
print(sorted(elf.values())[-1])


print("Part 2")
print(sum(sorted(elf.values())[-3:]))

